cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -0,0 +1,378 @@
+				artifactId: "spring-boot-starter-data-redis",
+			},
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-redis-reactive",
+			},
+		},
+	},
+	{
+		databaseDep: DbMongo,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-mongodb",
+			},
+			{
+				groupId:    "org.springframework.boot",
+				artifactId: "spring-boot-starter-data-mongodb-reactive",
+			},
+		},
+	},
+	{
+		databaseDep: DbCosmos,
+		mavenDependencies: []MavenDependency{
+			{
+				groupId:    "com.azure.spring",
+				artifactId: "spring-cloud-azure-starter-data-cosmos",
+			},
+		},
+	},
+}
+
+func updateProjectByAnalyzingSpringBootProject(
+	parentProject *mavenProject, mavenProject *mavenProject, azdProject *Project) {
+	if !isSpringBootApplication(mavenProject) {
+		log.Printf("Skip analyzing spring boot project. path = %s.", mavenProject.path)
+		return
+	}
+	var springBootProject = SpringBootProject{
+		springBootVersion:     detectSpringBootVersion(parentProject, mavenProject),
+		applicationProperties: readProperties(azdProject.Path),
+		parentProject:         parentProject,
+		mavenProject:          mavenProject,
+	}
+	addDatabases(azdProject, &springBootProject)
+	addServiceBus(azdProject, &springBootProject)
+	addEventHubs(azdProject, &springBootProject)
+	addStorageAccount(azdProject, &springBootProject)
+	addSpringCloudAzure(azdProject, &springBootProject)
+}
+
+func addDatabases(azdProject *Project, springBootProject *SpringBootProject) {
+	databaseDepMap := map[DatabaseDep]struct{}{}
+	for _, rule := range databaseDependencyRules {
+		for _, targetDependency := range rule.mavenDependencies {
+			for _, projectDependency := range springBootProject.mavenProject.Dependencies {
+				if projectDependency.GroupId == targetDependency.groupId &&
+					projectDependency.ArtifactId == targetDependency.artifactId {
+					databaseDepMap[rule.databaseDep] = struct{}{}
+					logServiceAddedAccordingToMavenDependency(rule.databaseDep.Display(),
+						projectDependency.GroupId, projectDependency.ArtifactId)
+					break
+				}
+			}
+		}
+	}
+	if len(databaseDepMap) > 0 {
+		azdProject.DatabaseDeps = slices.SortedFunc(maps.Keys(databaseDepMap),
+			func(a, b DatabaseDep) int {
+				return strings.Compare(string(a), string(b))
+			})
+	}
+}
+
+func logServiceAddedAccordingToMavenDependency(resourceName, groupId string, artifactId string) {
+	logServiceAddedAccordingToMavenDependencyAndExtraCondition(resourceName, groupId, artifactId, "")
+}
+
+func logServiceAddedAccordingToMavenDependencyAndExtraCondition(
+	resourceName, groupId string, artifactId string, extraCondition string) {
+	insertedString := ""
+	extraCondition = strings.TrimSpace(extraCondition)
+	if extraCondition != "" {
+		insertedString = " and " + extraCondition
+	}
+	log.Printf("Added '%s' because found dependency '%s:%s'%s.",
+		resourceName, groupId, artifactId, insertedString)
```

Comment:
Use one entry to print related log, which can make it easier to keep all related log have same format.